src/components/PostDetails.tsx

```diff
@@ -1,118 +1,135 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { deleteComment, getComments } from '../utils/api';
+
+interface Props {
+  post: Post,
+}
+
+export const PostDetails: React.FC<Props> = React.memo(({ post }) => {
+  const [commentsFromServer, setComments] = useState<Comment[] | null>(null);
+  const [isLoading, setLoading] = useState(false);
+  const [hasOpenForm, setOpenForm] = useState(false);
+  const [loadingErrorNotice, setLoadingErrorNotice] = useState('');
+  const hasLoadingError = !!loadingErrorNotice;
```

Comment:
couldn't you just use `!!loadingErrorNotice` instead of creating a new variable? it looks a little redundant